test/tools/ossfuzz/protoToYul.cpp

```diff
@@ -1138,16 +1250,36 @@ void ProtoConverter::visit(Statement const& _x)
 void ProtoConverter::openBlockScope()
 {
 	m_scopeFuncs.push_back({});
+
 	// Create new block scope inside current function scope
 	if (m_inFunctionDef)
+	{
+		yulAssert(
+			!m_funcVars.empty(),
+			"Proto fuzzer: Invalid data structure"
+		);
 		m_funcVars.back().push_back(vector<string>{});
+		if (m_inForInitScope && m_forInitScopeExtEnabled)
+		{
+			yulAssert(
+				!m_funcForLoopInitVars.empty(),
+				"Proto fuzzer: Invalid data structure"
+			);
+			m_funcForLoopInitVars.back().push_back(vector<string>{});
+		}
+	}
 	else
-		m_globalVars.push_back(vector<string>{});
+	{
+		m_globalVars.push_back({});
+		if (m_inForInitScope && m_forInitScopeExtEnabled)
+			m_globalForLoopInitVars.push_back(vector<string>{});
+	}
```

Comment:
When we are entering a for-init block either from function or global scope, we create a 1D vector to hold any variables that may subsequently be declared.